azure/services/agentpools/spec.go

```diff
@@ -364,6 +379,10 @@ func (s *AgentPoolSpec) Parameters(ctx context.Context, existing interface{}) (p
 		},
 	}
 
+	if s.SpotMaxPrice != nil {
```

Comment:
Let's move this above where the other optional configs get set (see `sku` for example). 

Also, should we do the same thing for `ScaleDownMode`? since they are both pointers